### ISSUE TITLE
Segmentation fault while executing inside docker `CodeCompass_parser --help`

### ISSUE THREAD

**Issue by oleg-work** [Ericsson/CodeCompass#495]
So a segmentation fault.
Some basic info:
```
.CodeCompass ☺  git show
commit 1097e5ce0519278dffe44c97e57ed4120d96e2e7 (HEAD -> master, origin/master, origin/HEAD)
Merge: 423652d 93078cd
Author: Máté Cserép <mcserep@gmail.com>
Date:   Sat Nov 21 12:27:36 2020 +0100

    Merge pull request #493 from mcserep/circular_dep_fix
    
    Incremental parsing of projects with circular dependencies

.CodeCompass ☺  uname -a
Linux thankful_user-lcl 5.4.0-54-generic #60~18.04.1-Ubuntu SMP Fri Nov 6 17:25:16 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

.CodeCompass ☺  tree -d -L 2
.
├── build
│   ├── CMakeFiles
│   ├── logger
│   ├── model
│   ├── parser
│   ├── plugins
│   ├── scripts
│   ├── service
│   ├── Testing
│   ├── util
│   ├── webgui
│   └── webserver
├── CodeCompass
│   ├── doc
│   ├── docker
│   ├── lib
│   ├── logger
│   ├── model
│   ├── parser
│   ├── plugins
│   ├── scripts
│   ├── service
│   ├── util
│   ├── webgui
│   └── webserver
├── install
│   ├── bin
│   ├── lib
│   └── share
├── test_workspace
└── workspace
    └── cppcheck
```
I also attached everything that happened in bash in codecompass.log including launch with gdb.
[codecompass.log](https://github.com/Ericsson/CodeCompass/files/5616169/codecompass.log). Also if I run, BUILD_TYPE=Debug:
```
.CodeCompass ☺  docker run --rm -ti   --env DATABASE=sqlite --env BUILD_TYPE=Debug   --volume /home/lil-peppa/Apps/CodeCompass:/CodeCompass --volume /home/lil-peppa/Apps/cppcheck:/projects/cppcheck -p 8001:8080 codecompass:dev /bin/bash
```
problem disappears. Will try to investigate by myself, but not sure when I ll get spare time to do this.

**Comment by mcserep** [comment]
Hello oleg,

Thanks for reporting this issue. I think this is the same as #405 and the fix (see #490) have been merged into master a few days ago. Can you update your codebase and test whether the issue is gone?

Kind regards,
Máté

**Comment by oleg-work** [comment]
@mcserep

Sorry I forgot greeting, hallo!

I updated, seems like problem is gone.

Thanks,
Oleg
